src/rpc/blockchain.cpp

```diff
@@ -1515,8 +1514,18 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
         }
     }
 
-    if (blockcount < 1 || blockcount >= pindex->nHeight) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid block count: should be between 1 and the block's height");
+    if (pindex->nHeight < 2) {
+        blockcount = 0;
+    } else {
+        blockcount = std::min(blockcount, pindex->nHeight - 1);
+    }
+
+    if (!request.params[0].isNull()) {
+        blockcount = request.params[0].get_int();
+
+        if (blockcount < 1 || blockcount >= pindex->nHeight) {
```

Comment:
I think the reason to disallow 0 is that this RPC was originally intended to be able to calculate `txrate`.
Therefore if user specifies 0 by parameter, PRC should returns illegal parameter error.
So, for only genesis block and 1st block, I think it can allow 0 internally when user is not specify block count in 1st parameter, and can avoid error to be occured.

But now, it is easier to understand if we allow 0.